internal/pkg/earpointer/analysis.go

```diff
@@ -46,17 +56,30 @@ var Analyzer = &analysis.Analyzer{
 // for all reachable contexts for that function. Both the intra-procedural
 // instructions and inter-procedural instructions are handled.
 type visitor struct {
-	state    *state     // mutable state
-	contexts []*Context // for context sensitive analysis
+	state    *state                              // mutable state
+	contexts map[*ssa.Function][]*Context        // for context sensitive analysis
+	callees  map[*ssa.CallCommon][]*ssa.Function // callee functions at each callsite
+	contextK int
 }
 
 func run(pass *analysis.Pass) (interface{}, error) {
 	ssainput := pass.ResultOf[buildssa.Analyzer].(*buildssa.SSA)
-	vis := visitor{state: NewState(), contexts: []*Context{&emptyContext}}
-	vis.initGlobalReferences(ssainput)
+	p := analyze(ssainput.Pkg)
+	return p, nil
+}
+
+// Analyzes an SSA program and build the partition information.
+func analyze(pkg *ssa.Package) *Partitions {
+	prog := pkg.Prog
```

Comment:
Consider adding validation that `K=0` or `K=1` here, aborting early otherwise.

Reply:
Now it supports any K value. The outdated comments have been deleted.

Reply:
Oh, I missed the comments.  I was looking at
```golang
func init() {
	Analyzer.Flags.IntVar(&contextK, "contextK", 0,
		`the K value (default=0) in context sensitivity. Currently only K=0 or K=1 are supported`)
}
```

So the help string will need to be updated.